be/src/vec/exec/scan/vfile_scanner.cpp

```diff
@@ -491,20 +491,34 @@ Status VFileScanner::_get_next_reader() {
             ParquetReader* parquet_reader = new ParquetReader(
                     _profile, _params, range, _state->query_options().batch_size,
                     const_cast<cctz::time_zone*>(&_state->timezone_obj()), _io_ctx.get());
+            parquet_reader->open();
             if (!_is_load && _push_down_expr == nullptr && _vconjunct_ctx != nullptr) {
                 RETURN_IF_ERROR(_vconjunct_ctx->clone(_state, &_push_down_expr));
                 _discard_conjuncts();
             }
-            init_status = parquet_reader->init_reader(_file_col_names, _colname_to_value_range,
-                                                      _push_down_expr);
             if (range.__isset.table_format_params &&
                 range.table_format_params.table_format_type == "iceberg") {
+                _table_col_to_file_col.clear();
```

Comment:
I think these logics can be put into `iceberg reader`? Because it is only used for iceberg?